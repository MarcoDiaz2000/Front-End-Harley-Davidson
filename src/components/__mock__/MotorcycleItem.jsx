import React from 'react';

const MotorcycleItem = () => (
  <li className="flex flex-col sm:flex-row justify-between items-center p-2 my-2 rounded-lg bg-white white-shadow-i">
    <span className="font-roboto font-bold text-black">Motorcycle</span>
    <button
      type="button"
      className="p-2 rounded hover:bg-customDark-900 font-roboto font-bold text-white bg-harleyOrange"
    >
      Delete
    </button>
  </li>
);
export default MotorcycleItem;
