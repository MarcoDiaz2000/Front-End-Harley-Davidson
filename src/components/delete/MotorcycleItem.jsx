import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { toast } from 'react-toastify';
import { markBikeAsRemoved, markBikeAsRestored } from '../../redux/bikes/thunk';

const MotorcycleItem = ({ motorcycle }) => {
  const dispatch = useDispatch();

  const handleDelete = (id) => {
    dispatch(markBikeAsRemoved({ id }));
    toast.success('Motorcycle marked for deletion');
  };

  const handleRecover = (id) => {
    dispatch(markBikeAsRestored({ id }));
    toast.success('Motorcycle recovered');
  };

  return (
    <li className={`flex flex-col sm:flex-row justify-between items-center p-2 my-2 rounded-lg transform transition duration-500 hover:scale-105 ${motorcycle.removed ? 'bg-harleyOrange white-shadow-i' : 'bg-white white-shadow-i'}`}>
      <span className={`font-roboto font-bold ${motorcycle.removed ? 'text-white' : 'text-black'}`}>{motorcycle.name}</span>
      <button
        type="button"
        onClick={() => (motorcycle.removed
          ? handleRecover(motorcycle.id)
          : handleDelete(motorcycle.id))}
        className={`p-2 rounded hover:bg-customDark-900 font-roboto font-bold ${motorcycle.removed ? 'text-harleyOrange bg-white' : 'text-white bg-harleyOrange'}`}
      >
        {motorcycle.removed ? 'Recover' : 'Delete'}
      </button>
    </li>
  );
};

MotorcycleItem.propTypes = {
  motorcycle: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    removed: PropTypes.bool.isRequired,
  }).isRequired,
};

export default MotorcycleItem;
