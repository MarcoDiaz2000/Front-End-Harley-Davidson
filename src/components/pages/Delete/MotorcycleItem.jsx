import React from 'react';
import { useDispatch } from 'react-redux';
import { markMotorcycleAsRemoved, recoverMotorcycle } from '../redux/motorcycles/motorcycles';
import { toast } from 'react-toastify';

const MotorcycleItem = ({ motorcycle }) => {
  const dispatch = useDispatch();
  
  const handleDelete = (id) => {
    dispatch(markMotorcycleAsRemoved(id));
    toast.success('Motorcycle marked for deletion');
  };

  const handleRecover = (id) => {
    dispatch(recoverMotorcycle(id));
    toast.success('Motorcycle recovered');
  };

  return (
    <li className={`flex justify-between items-center p-2 my-2 rounded ${motorcycle.removed ? 'bg-red-200' : 'bg-white'}`}>
      <span>{motorcycle.name}</span>
      <button
        type="button"
        onClick={() => (motorcycle.removed ? handleRecover(motorcycle.id) : handleDelete(motorcycle.id))}
        className="bg-white text-green-500 p-2 rounded hover:bg-green-200"
      >
        {motorcycle.removed ? 'Recover' : 'Delete'}
      </button>
    </li>
  );
};

export default MotorcycleItem;
