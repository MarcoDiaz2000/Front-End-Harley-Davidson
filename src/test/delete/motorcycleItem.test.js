import React from 'react';
import renderer from 'react-test-renderer';
import MotorcycleItem from '../../components/__mock__/MotorcycleItem';

describe('MotorcycleItem', () => {
  it('renders correctly', () => {
    const component = renderer.create(<MotorcycleItem />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
