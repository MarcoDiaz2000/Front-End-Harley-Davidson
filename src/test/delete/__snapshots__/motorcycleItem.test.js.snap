// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MotorcycleItem renders correctly 1`] = `
<li
  className="flex flex-col sm:flex-row justify-between items-center p-2 my-2 rounded-lg bg-white white-shadow-i"
>
  <span
    className="font-roboto font-bold text-black"
  >
    Motorcycle
  </span>
  <button
    className="p-2 rounded hover:bg-customDark-900 font-roboto font-bold text-white bg-harleyOrange"
    type="button"
  >
    Delete
  </button>
</li>
`;
